### ISSUE TITLE
Explicit method group conversion suppresses unnecessary qualification warning

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#1208]
```C#
using System;

class X {
	void M() { R((Action)this.M); }
	void R(Action a) { }
}
```

`(Action)` is highlighted as unnecessary, but `this.` isn't.
Removing `(Action)` correctly highlights it in gray.

Typename qualifiers for static methods are similarly missed, but namespace qualifiers are not.

**Comment by SLaks** [comment]
Also, when rename inserts a cast to a delegate type to resolve a conflict, it will unnecessarily qualify the method group reference.
I assume that's the same underlying bug.
